modules/foundation/service.go

```
@@ -0,0 +1,155 @@
+	for _, user := range s.users {
+		if user.Email == email {
+			return NewConflictError("email already exists")
+		}
+	}
+	return nil
+}
+
+// generateID generates a simple random ID for demonstration
+func generateID() string {
+	bytes := make([]byte, 8)
+	rand.Read(bytes)
```

golangci-lint (1.64.8)

Error return value of `rand.Read` is not checked

(errcheck)